### PR TITLE
Add statically typed methods for configuring plugin with Kotlin

### LINKED ISSUE #406: Provide static accessor functions for javaApplication extension on DockerExtension
Follow On discussion to #385 as I can't re-open that issue.

I came up with a far better solution for providing the static accessor functions than simply documenting it in the readme. I implemented this solution successfully in the JUnit 5 gradle plugin.

The pr:
https://github.com/junit-team/junit5/pull/904

The commit:
https://github.com/junit-team/junit5/commit/4a2e8e04ecf054c01841c5f863d8ca5af9acd0d9

If you look over the PR you can see that this is the solution that [Stefan Oehme](https://github.com/oehme) (Gradle project maintainer) supports.

I propose adding the following method to the `DockerExtension`:

```groovy
    void javaApplication(org.gradle.api.Action<DockerJavaApplication> closure) {
        closure.execute(getProperty("javaApplication") as DockerJavaApplication)
    }
```

While in there it may be nice to add static methods for `DockerExistingContainer.targetContainerId` and `DockerCreateContainer.targetImageId`. This would make it so that the README section added in #388 could be removed.

### PR DESCRIPTION
**Pull request by JLLeitschuh**
Closes #406

### CODE REVIEW

**Comment by cdancy** [comment]
LGTM. @orzeh second pair of eyes for review?

File: src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplicationPlugin.groovy [hunk 1]
```diff
@@ -18,6 +18,7 @@ package com.bmuschko.gradle.docker
 import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
 import com.bmuschko.gradle.docker.tasks.image.DockerPushImage
 import com.bmuschko.gradle.docker.tasks.image.Dockerfile
+import groovy.transform.PackageScope
```
Comment (orzeh) [inline src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplicationPlugin.groovy]: unused import?

Comment (JLLeitschuh) [inline src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplicationPlugin.groovy]: Good catch. I would expect that there would be some sort of style linter that would have caught that for me like the [spotless](https://github.com/diffplug/spotless/tree/master/plugin-gradle) plugin. I'll go and remove it.

**Comment by cdancy** [comment]
@JLLeitschuh merged. Thanks for the PR!

**Pull request merged**